src/Plugin/EntityReferenceSelection/OgSelection.php

```diff
@@ -77,37 +139,45 @@ protected function buildEntityQuery($match = NULL, $match_operator = 'CONTAINS')
       $query->condition($bundle_key, $bundles, 'IN');
     }
 
-    $user_groups = $this->getUserGroups();
-    if (!$user_groups) {
+    // Get the identifier key of the entity.
+    $identifier_key = $definition->getKey('id');
+
+    if ($this->currentUser->isAnonymous()) {
```

Comment:
I think for now we can drop the anonymous support, and concentrate on the 99% use case -- auth users